### DISCUSSION TITLE
Need to stream external webCam on ubuntu 20.04.5 LTS (Focal Fossa)

### DISCUSSION THREAD

**Discussion by prshtv** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1887] [Q&A]
Dear Smart Folks,

I have installed the webRTC package as per the instructions in this github on an ubuntu host -20.04.5 LTS (Focal Fossa)
I have connected an external webCam to the Ubuntu Host and tested the camera and it works fine.
Now, I am trying to use the Sample: kvsWebrtcClientMasterGstSample and expecting that I would receive a stream of my webcam video on the kinesis. But I am not seeing any stream on Kinesis when I am tried to view via the media-playback option

**Link of the sample being used**:
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c?tab=readme-ov-file#sample-kvswebrtcclientmastergstsample

**Invocation command**: ./samples/kvsWebrtcClientMasterGstSample    MySignalingchannel   video-only     devicesrc

Can you please provide me some inputs how to stream my webCam to Kinesis? Am I in the right direction?

**Comment by stefankiesz** [comment]
Hi, can you confirm that you are using the media playback viewer in the "Signaling channels" section and not the "Video streams" section. Signaling channels will be the place to check for incoming footage for this use case. If that wasn't the issue, please try the following command and check if there is footage:
```
./samples/kvsWebrtcClientMasterGstSample MySignalingchannel video-only testsrc
```

**Comment by prshtv** [comment]
Hi, Yes I am using the signaling channel. I also tried to use the gst-launch-1.0 tool to test my pipeline

gst-launch-1.0 v4l2src do-timestamp=TRUE device=/dev/video0 ! videoconvert ! video/x-raw,format=YUY2,width=640,height=480,framerate=30/1 ! x264enc  bframes=0 key-int-max=45 bitrate=500 ! video/x-h264,stream-format=avc,alignment=au,profile=baseline ! kvssink stream-name="kvs-sig-chan-1" storage-size=512 access-key="<My-Access-ID>" secret-key="<My-secret>" aws-region="us-west-2"

I am getting this issue and then no streaming happens
WARNING: erroneous pipeline: could not link videoconvert0 to x264enc0, x264enc0 can't handle caps video/x-raw, format=(string)YUY2, width=(int)640, height=(int)480, framerate=(fraction)30/1

Am I using the command correctly?

**Comment by prshtv** [comment]
Hi, BTW while I invoke this command which you suggested me. It shows a moving dot on KVS signaling channel console and it works fine
./samples/kvsWebrtcClientMasterGstSample MySignalingchannel video-only testsrc

**Comment by stefankiesz** [comment]
You are using that command correctly, but the YUY2 format is not supported by GStreamer's x264enc element. Please refer to [GStreamer documentation](https://gstreamer.freedesktop.org/documentation/x264/index.html?gi-language=c#:~:text=format%3A%20%7B%20Y444%2C%20Y42B%2C%20I420%2C%20YV12%2C%20NV12%2C%20GRAY8%2C%20Y444_10LE%2C%20I422_10LE%2C%20I420_10LE%20%7D) for element compatibility details.

Using the [Encode and stream video from a USB camera on Ubuntu example](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/examples-gstreamer-plugin.html) as reference, can you try changing the format to I420 from YUY2?

And for the sample run, can you please provide verbose level logs?
